jupyter-spark/extensions/spark.js

```diff
@@ -24,17 +35,26 @@ var update_cache = function(callbacks) {
     $.getJSON(API + '/applications').done(function(applications) {
         var num_applications = cache.length;
         var num_completed = 0;
-        applications.forEach(function(application, i) {
-            $.getJSON(API + '/applications/' + application.id + '/jobs').done(function (jobs) {
-                cache[i] = application;
-                cache[i].jobs = jobs;
-
-                num_completed++;
-                if (num_completed === num_applications && cbs) {
-                    cbs.fire(cache);
-                }
+        // Check if Spark is running before processing applications
+        if(!applications.hasOwnProperty('error')){
```

Comment:
should set `spark_is_running = false` if it *does* have the error property (ie. if Spark crashes or is shut down)